### PR TITLE
DOC: Add line chart with markers example

### PR DESCRIPTION
**Pull request by eitanlees**
Added a simple example of a line chart with markers discussed in #876 

I can also update 

- [Connected Scatterplot (Lines with Custom Paths)](https://altair-viz.github.io/gallery/connected_scatterplot.html)

- [Multiple Marks](https://altair-viz.github.io/gallery/multiple_marks.html)

if you think using `mark_line(point=True)` is preferred to the layering method.

### CODE REVIEW

**Comment by kanitw** [comment]
> mark_line(point=True)

is better because the point markers will have opacity = 1 be default.  (The layering method will make point's opacity a bit semi-transparent like in scatterplots.)

**Comment by jakevdp** [comment]
Yes, it would be great to update those other examples.

**Comment by jakevdp** [comment]
There's also a couple instances in the sphinx docs that would be worth updating as well

**Comment by eitanlees** [comment]
I fixed the two examples listed earlier. Now for the sphinx docs. 

I see one example in the [Layered Charts](https://altair-viz.github.io/user_guide/compound_charts.html#layered-charts) section. 

Any others?

File: altair/vegalite/v2/examples/connected_scatterplot.py [hunk 1]
```diff
@@ -2,23 +2,16 @@
 Connected Scatterplot (Lines with Custom Paths)
 -----------------------------------------------
 
-This example shows how layering can be used to build a plot. This dataset tracks miles driven per capita along with gas prices annually from 1956 to 2010. It is based on the May 2, 2010 New York Times article 'Driving Shifts Into Reverse'. See http://mbostock.github.io/protovis/ex/driving.html .
+This example show how the order encoding can be used to draw a custom path. The dataset tracks miles driven per capita along with gas prices annually from 1956 to 2010. It is based on Hannah Fairfield's article 'Driving Shifts Into Reverse'. See https://archive.nytimes.com/www.nytimes.com/imagepages/2010/05/02/business/02metrics.html for the original.
```
Comment (jakevdp) [inline altair/vegalite/v2/examples/connected_scatterplot.py]: Can you break this line?

**Comment by jakevdp** [comment]
Looks good aside from the one long line. Thanks!

File: altair/vegalite/v2/examples/connected_scatterplot.py (hunk 1 above)
Comment (eitanlees) [reply altair/vegalite/v2/examples/connected_scatterplot.py]: Sure thing

**Comment by jakevdp** [comment]
Looks great, thanks!

**Pull request merged**